.storybook/webpack.config.js

```diff
@@ -6,34 +6,61 @@
 const path = require('path')
 const webpack = require('webpack')
 const fs = require('fs')
+const config = require('../build/commands/lib/config')
+const genTsConfig = require('../build/commands/lib/genTsConfig')
 const {
   fallback,
   provideNodeGlobals
 } = require('../components/webpack/polyfill')
-
-const buildConfigs = ['Component', 'Static', 'Debug', 'Release']
-const extraArchitectures = ['arm64', 'x86']
-
-function getBuildOuptutPathList(buildOutputRelativePath) {
+const generatePathMap = require('../components/webpack/path-map')
+
+// Choose which brave-core build directory to look for pre-compiled
+// resource dependencies:
+// 1. Default for local builds for the actual platform / architecture
+// 2. platform / architecture overriden by environment variables
+// 3. most recently built - this caters to the common scenario when a
+// non-standard target has been built but no arguments are provided to storybook
+config.update({
+  target_arch: process.env.TARGET_ARCH,
```

Comment:
IT will support whatever we currently support for our other commands. If more recent versions of npm don't allow custom params then you must use what we already support. What most devs and CI use is CLI params to `npm run sync` and `npm run build`. However, Storybook doesn't support custom arguments, so for this command we specify via environment param instead.